brave-okhttp/src/main/java/com/github/kristofa/brave/okhttp/BraveOkHttpRequestResponseInterceptor.java

```diff
@@ -29,10 +33,22 @@ public Response intercept(Interceptor.Chain chain) throws IOException {
     Request request = chain.request();
     Request.Builder builder = request.newBuilder();
     OkHttpRequest okHttpRequest = new OkHttpRequest(builder, request);
-    clientRequestInterceptor.handle(new HttpClientRequestAdapter(okHttpRequest, spanNameProvider));
+    clientRequestInterceptor.handle(createRequestAdapter(okHttpRequest));
     Response response = chain.proceed(builder.build());
-    clientResponseInterceptor.handle(new HttpClientResponseAdapter(new OkHttpResponse(response)));
+    clientResponseInterceptor.handle(createResponseAdapter(response));
     return response;
   }
 
+  protected SpanNameProvider getSpanNameProvider() {
```

Comment:
Consider the caller can control the constructor, I think this part is over-extension.